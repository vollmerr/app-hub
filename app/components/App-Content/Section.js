import styled from 'styled-components';

import theme from 'utils/theme';

const Section = styled.div`
  padding: ${theme.hub.padding}px;
  background: ${theme.white};
  box-shadow: 0 0 3px ${theme.neutralLight};
`;

export default Section;
