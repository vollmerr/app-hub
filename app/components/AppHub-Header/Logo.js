import React from 'react';
import styled from 'styled-components';

import theme from 'utils/theme';

import Link from './Link';
import odiLogo from './odiLogo.png';

const Img = styled.img`
  height: calc(${theme.hub.headerHeight} - 10px);
  margin:  0 0 0 -10px;
`;

const Logo = () => (
  <Link href={'http://www.cdt.ca.gov'}>
    <Img
      src={odiLogo}
      alt={'CDT ODI Logo'}
    />
  </Link>
);

export default Logo;
