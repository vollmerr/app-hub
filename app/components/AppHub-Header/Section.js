import styled from 'styled-components';

const Section = styled.div`
  display: flex;
`;

export default Section;
