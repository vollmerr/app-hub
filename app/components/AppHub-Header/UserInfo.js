import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import theme from 'utils/theme';

import Item from './Item';

export const Name = styled.span`
  color: ${theme.themeLighterAlt};
  line-height: ${theme.hub.headerHeight}px;
  display: inline-block;
  font-family: "SegoeUI-SemiLight-final","Segoe UI SemiLight","Segoe UI WPC Semilight","Segoe UI",Segoe,Tahoma,Helvetica,Arial,sans-serif;
  font-size: 18px;
  -webkit-font-smoothing: antialiased;
`;

export const Initials = styled.div`
  width: calc(${theme.hub.headerHeight}px - 10px);
  height: calc(${theme.hub.headerHeight}px - 10px);
  color: ${theme.themeLighterAlt};
  background: ${theme.themePrimary};
  border-radius: 50%;
  margin: 0 15px 0 10px;
  display: flex;
  justify-content: center;
  align-items: center;
`;


const UserInfo = ({ name, initials }) => (
  <Item>
    <Name>{name}</Name>
    <Initials>{initials}</Initials>
  </Item>
);


const { string } = PropTypes;

UserInfo.propTypes = {
  name: string.isRequired,
  initials: string.isRequired,
};

export default UserInfo;
