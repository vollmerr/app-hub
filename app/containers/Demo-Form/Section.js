import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import theme from 'utils/theme';

const Wrapper = styled.div`
  flex: 30%;
  min-width: ${theme.breakpoints.xs}px;
  padding: 15px;
  margin: 5px;
  background: ${theme.white};
`;


export class Section extends React.PureComponent {
  render() {
    const { title, allNames, byName } = this.props;

    return (
      <Wrapper>
        <h3>{title}</h3>
        {
          // loop through field names for section (in order specified)
          allNames.map((name) => {
            // pull out component from other props (must be capitalized so rename to 'C')
            const { component: C, ...props } = byName[name];
            // render individual field with props (required, label, etc)
            return (
              <C {...props} key={name} />
            );
          })
        }
      </Wrapper>
    );
  }
}

Section.propTypes = {
  title: PropTypes.string.isRequired,
  allNames: PropTypes.array.isRequired,
  byName: PropTypes.object.isRequired,
};

export default Section;
