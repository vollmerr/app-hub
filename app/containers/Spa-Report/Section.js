import styled from 'styled-components';

import theme from 'utils/theme';


const Section = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  width: 100%;
  min-height: 300px;

  @media (min-width: ${theme.breakpoints.sm}px) {
    min-width: ${theme.breakpoints.sm - 40}px;
  }
`;

export default Section;
