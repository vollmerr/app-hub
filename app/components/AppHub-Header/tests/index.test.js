import React from 'react';
import { shallow } from 'enzyme';

import {
  HELP_PANEL,
  APPS_PANEL,
  APP_NAV_PANEL,
  ALERTS_PANEL,
  DEV_PANEL,
} from 'containers/AppHub/constants';

import AppHubHeader from '../index';
import Section from '../Section';
import UserInfo from '../UserInfo';
import Logo from '../Logo';

global.isDev = false;

const props = {
  isMobile: false,
  onClick: jest.fn(),
  appName: 'test app name',
  appPath: '/test-route',
  panel: 'test panel name',
  isOpen: false,
  userName: 'test user',
  alerts: [{ app: 'app name 1', message: 'message 1' }],
};

const testIfChecked = (wrapper, panel) => {
  wrapper.setProps({ panel, isOpen: true });
  expect(wrapper.find({ panel }).prop('checked')).toEqual(true);
};


describe('<AppHubHeader />', () => {
  describe('desktop view', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(
        <AppHubHeader {...props} />
      );
    });

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render two sections (left and right)', () => {
      expect(wrapper.find(Section).length).toEqual(2);
    });

    it('should not render app nav link', () => {
      const panel = APP_NAV_PANEL;
      expect(wrapper.find({ panel }).length).toEqual(0);
    });

    it('should render the logo', () => {
      expect(wrapper.find(Logo).length).toEqual(1);
    });

    it('should render the App Hub title/icon', () => {
      const text = 'App Hub';
      expect(wrapper.find({ text }).length).toEqual(1);
    });

    it('should render the app title if passed an `appName`', () => {
      const text = props.appName;
      expect(wrapper.find({ text }).length).toEqual(1);

      wrapper.setProps({ appName: undefined });
      expect(wrapper.find({ text }).length).toEqual(0);
    });

    it('should render the user information if passed a `userName`', () => {
      expect(wrapper.find(UserInfo).length).toEqual(1);

      wrapper.setProps({ userName: undefined });
      expect(wrapper.find(UserInfo).length).toEqual(0);
    });

    it('should generate the correct user intitials', () => {
      const initials = 'TU'; // for 'test user'
      expect(wrapper.find(UserInfo).prop('initials')).toEqual(initials);
    });

    it('should render the dev panel link if in development mode', () => {
      const panel = DEV_PANEL;
      expect(wrapper.find({ panel }).length).toEqual(0);

      global.isDev = true;
      wrapper = shallow(<AppHubHeader {...props} />);
      testIfChecked(wrapper, panel);
      global.isDev = false; // reset to not dev mode for other tests.
      expect(wrapper.find({ panel }).length).toEqual(1);
    });

    it('should render the alerts panel link if passed `alerts`', () => {
      const panel = ALERTS_PANEL;
      expect(wrapper.find({ panel }).length).toEqual(1);

      testIfChecked(wrapper, panel);

      wrapper.setProps({ alerts: [] });
      expect(wrapper.find({ panel }).length).toEqual(0);
    });

    it('should render the help panel link', () => {
      const panel = HELP_PANEL;
      expect(wrapper.find({ panel }).length).toEqual(1);

      testIfChecked(wrapper, panel);
    });

    it('should render the App Hub navigation link', () => {
      const panel = APPS_PANEL;
      expect(wrapper.find({ panel }).length).toEqual(1);

      testIfChecked(wrapper, panel);
    });
  });

  describe('mobile view', () => {
    let wrapper;
    beforeEach(() => {
      const mobileProps = { ...props, isMobile: true };
      wrapper = shallow(
        <AppHubHeader {...mobileProps} />
      );
    });

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render two sections (left and right)', () => {
      expect(wrapper.find(Section).length).toEqual(2);
    });

    it('should render app nav link if passed a `appName`', () => {
      const panel = APP_NAV_PANEL;
      expect(wrapper.find({ panel }).length).toEqual(1);

      testIfChecked(wrapper, panel);

      wrapper.setProps({ appName: undefined });
      expect(wrapper.find({ panel }).length).toEqual(0);
    });

    it('should not render the logo', () => {
      expect(wrapper.find(Logo).length).toEqual(0);
    });

    it('should not render the App Hub title/icon', () => {
      const text = 'App Hub';
      expect(wrapper.find({ text }).length).toEqual(0);
    });

    it('should render the app title if passed one', () => {
      const text = props.appName;
      expect(wrapper.find({ text }).length).toEqual(1);

      wrapper.setProps({ appName: undefined });
      expect(wrapper.find({ text }).length).toEqual(0);
    });

    it('should not render the user information', () => {
      expect(wrapper.find(UserInfo).length).toEqual(0);
    });

    it('should render the dev panel link if in development mode', () => {
      const panel = DEV_PANEL;
      expect(wrapper.find({ panel }).length).toEqual(0);

      global.isDev = true;
      wrapper = shallow(<AppHubHeader {...props} />);
      testIfChecked(wrapper, panel);
      global.isDev = false; // reset to not dev mode for other tests.
      expect(wrapper.find({ panel }).length).toEqual(1);
    });

    it('should render the alerts panel link if passed `alerts`', () => {
      const panel = ALERTS_PANEL;
      expect(wrapper.find({ panel }).length).toEqual(1);

      testIfChecked(wrapper, panel);

      wrapper.setProps({ alerts: [] });
      expect(wrapper.find({ panel }).length).toEqual(0);
    });

    it('should render the help panel link', () => {
      const panel = HELP_PANEL;
      expect(wrapper.find({ panel }).length).toEqual(1);

      testIfChecked(wrapper, panel);
    });

    it('should render the App Hub navigation link', () => {
      const panel = APPS_PANEL;
      expect(wrapper.find({ panel }).length).toEqual(1);

      testIfChecked(wrapper, panel);
    });
  });
});
