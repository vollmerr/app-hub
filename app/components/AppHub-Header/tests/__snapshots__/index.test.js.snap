// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppHubHeader /> desktop view should render correctly 1`] = `
<Wrapper__Wrapper>
  <Section__Section>
    <Logo />
    <Line__Line
      partial={true}
    />
    <Link
      iconProps={
        Object {
          "iconName": "Apphub",
          "style": Object {
            "fontSize": "24px",
            "marginRight": "10px",
          },
        }
      }
      onClick={[Function]}
      padding="15px"
      text="App Hub"
      title="App Hub"
      to="/"
    />
    <Line__Line
      partial={true}
    />
    <Link
      onClick={[Function]}
      padding="15px"
      text="test app name"
      title="test app name"
      to="/test-route"
    />
  </Section__Section>
  <Section__Section>
    <UserInfo
      initials="TU"
      name="test user"
    />
    <Line__Line />
    <Link
      checked={false}
      iconProps={
        Object {
          "iconName": "alertsMenu",
        }
      }
      onClick={[Function]}
      panel="app/AppHub/ALERTS_PANEL"
      title="Alerts Panel"
    />
    <Line__Line />
    <Link
      checked={false}
      iconProps={
        Object {
          "iconName": "helpMenu",
        }
      }
      onClick={[Function]}
      panel="app/AppHub/HELP_PANEL"
      title="Help Panel"
    />
    <Line__Line />
    <Link
      checked={false}
      dark={true}
      iconProps={
        Object {
          "iconName": "appHubMenu",
        }
      }
      onClick={[Function]}
      panel="app/AppHub/APPS_PANEL"
      title="Hub Navigation"
    />
  </Section__Section>
</Wrapper__Wrapper>
`;

exports[`<AppHubHeader /> mobile view should render correctly 1`] = `
<Wrapper__Wrapper>
  <Section__Section>
    <Link
      checked={false}
      iconProps={
        Object {
          "iconName": "appMenu",
        }
      }
      onClick={[Function]}
      panel="app/AppHub/APP_NAV_PANEL"
      title="test app name Navigation"
    />
    <Line__Line
      partial={false}
    />
    <Link
      onClick={[Function]}
      padding="15px"
      text="test app name"
      title="test app name"
      to="/test-route"
    />
  </Section__Section>
  <Section__Section>
    <Line__Line />
    <Link
      checked={false}
      iconProps={
        Object {
          "iconName": "alertsMenu",
        }
      }
      onClick={[Function]}
      panel="app/AppHub/ALERTS_PANEL"
      title="Alerts Panel"
    />
    <Line__Line />
    <Link
      checked={false}
      iconProps={
        Object {
          "iconName": "helpMenu",
        }
      }
      onClick={[Function]}
      panel="app/AppHub/HELP_PANEL"
      title="Help Panel"
    />
    <Line__Line />
    <Link
      checked={false}
      dark={true}
      iconProps={
        Object {
          "iconName": "appHubMenu",
        }
      }
      onClick={[Function]}
      panel="app/AppHub/APPS_PANEL"
      title="Hub Navigation"
    />
  </Section__Section>
</Wrapper__Wrapper>
`;
